bundles/org.eclipse.jface/src/org/eclipse/jface/action/StatusLineManager.java

```diff
@@ -149,15 +149,29 @@ public IProgressMonitor getProgressMonitor() {
 				: new NullProgressMonitor();
 
 		return new IProgressMonitor() {
+			private volatile boolean taskStarted;
 
 			@Override
 			public void beginTask(String name, int totalWork) {
+				if (taskStarted) {
+					throw new IllegalStateException("beginTask must only be called once per instance"); //$NON-NLS-1$
```

Comment:
I think for this to become useful one should record the callstack of the first invocation to give a hint where this has originated.

The easiest way would be to construct a new Runtime exception e.g. with message "first invocation here" and use that as the cause of the exception